packages/admin-frontend/pages/admin/statistics/most-read.js

```diff
@@ -0,0 +1,180 @@
+// @flow
+
+/**
+ * Part of GDL gdl-frontend.
+ * Copyright (C) 2019 GDL
+ *
+ * See LICENSE
+ */
+
+import * as React from 'react';
+
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableRow,
+  TableHead,
+  TableFooter,
+  Typography,
+  CircularProgress,
+  Button,
+  TextField,
+  FormControl,
+  Grid
+} from '@material-ui/core';
+import { SaveAlt as SaveIcon } from '@material-ui/icons';
+import type { BookStatistics } from '../../../types';
+import Container from '../../../components/Container';
+import Layout from '../../../components/Layout';
+import { fetchMostReadBooks } from '../../../lib/fetch';
+import formatMostReadDataToObjects from '../../../lib/utils';
+
+type State = {
+  mostReadBooks: Array<BookStatistics>,
+  loading: boolean,
+  numberOfBooksWanted: number,
+  periodOfDays: number
+};
+class MostRead extends React.Component<*, State> {
+  state = {
+    mostReadBooks: [],
+    loading: true,
+    numberOfBooksWanted: 0,
+    periodOfDays: 0
+  };
+
+  async componentDidMount() {
+    this.fetchResults();
+  }
+
+  handleChange = (name: string) => (event: any) => {
+    this.setState({
+      [name]: event.target.value
+    });
+  };
+
+  fetchResults = async () => {
+    const { numberOfBooksWanted, periodOfDays } = this.state;
+    this.setState({ loading: true });
+    const bookStatisticsRes = await fetchMostReadBooks(
+      numberOfBooksWanted,
+      periodOfDays
+    );
+
+    if (bookStatisticsRes.isOk) {
+      this.setState({
+        loading: false,
+        mostReadBooks: formatMostReadDataToObjects(bookStatisticsRes).data
+      });
+    }
+  };
+
+  handleExportButtonClick = async () => {
+    const { numberOfBooksWanted, periodOfDays } = this.state;
+    const bookStatisticsRes = await fetchMostReadBooks(
```

Comment:
Her henter du den samma dataen som blir initielt hentet ved rendring av siden og den blir lagret i state. Det betyr dette blir et ekstra unødvendig kall som du kan slippe unna med? Du må potensielt ha logikk får å sjekke at det finnes data i `this.state.mostReadBooks`. Et forslag er å sette den som `undefined` i starten, da slikker du å sammenligne med et ``this.state.mostReadBooks.length`